src/txn/txn_timestamp.c

```diff
@@ -928,6 +928,8 @@ __wt_txn_set_read_timestamp(
 			did_roundup_to_oldest = true;
 		} else {
 			__wt_readunlock(session, &txn_global->rwlock);
+			if (!WT_VERBOSE_ISSET(session, WT_VERB_TIMESTAMP))
+				return (EINVAL);
```

Comment:
I was talking to Eric and Louis about this issue. This change rather a hack. At a minimum it needs a long comment about why we'd return EINVAL when the next line down is also returning EINVAL with more information and why does verbose factor into it at all? MongoDB is raising an error when receiving the original value and message, but they would prefer to have the actual content as an informational message I believe.

Reply:
What about something like:
```
/* comment about it */
__wt_msg( information about timestamps);
return (EINVAL);
```
that is not predicated on verbose and still gives the information. That is tolerable but not great. It is impossible to know in general, which of our messages like this one are "informational" errors.

Reply:
My preference is no message at all, simply `return (error-maybe-not-EINVAL);` and document that fact. In other words, what you're referring to as "informational errors" shouldn't be logged at all.

I'm not the go-to person on this stuff, but it seems reasonable to me that application read timestamps can age out, and it's OK when that happens.

I agree about the verbose check (I put in `WT_VERBOSE_ISSET` when I thought `test/format` was the only application that cared about this).

Reply:
I'm in half-agreement. I think from the library perspective, detecting this error is valid. I also agree that `error-not-EINVAL` is better. I just spoke with @milkie. It seems like changing `EINVAL` to `WT_ROLLBACK` (in the `WT_RET_MSG` call) may solve the problem. The caller to `timestamp_transaction` can retry with another timestamp and they do get the informational message (which is desired).

Reply:
I pushed a branch that discards the message and documents the error. At this point, maybe it's best if @bvpvamsikrishna, @agorrod, and @michaelcahill weigh in and make this call.

Reply:
@milkie has said to me, several times, that they still want the message. Is there a problem with the suggestion of changing EINVAL to WT_ROLLBACK in WT_RET_MSG?